parmed/amber/_amberparm.py

```diff
@@ -1996,6 +1998,31 @@ def _add_missing_13_14(self, ignore_inconsistent_vdw=False):
 
     #===================================================
 
+    def _get_atom_dict_for_alternate_labels(self, residues):
+        from collections import defaultdict
+
+        atom_dict = defaultdict(list)
+        for resid, residue in enumerate(residues):
+            atom_dict[resid] = defaultdict(list)
+            for atom in residue.atoms:
+                atom_dict[resid][atom.name].append(atom)
+        return atom_dict
```

Comment:
Move the import to the top of the file.  Also, `atom_dict` looks to me like it should not be a dict at all, but just a simple list:

```Python
atom_collection = [defaultdict(list) for r in residues]
for ad, res in zip(atom_collection, residues):
    for atom in res:
        ad[atom.name].append(atom)
return atom_collection
```

Also, you can use `self.residues` rather than passing in a `residues` object.